base-server.d.ts

```diff
@@ -0,0 +1,704 @@
+import { Log, Store, ServerConnection, TestTime } from '@logux/core'
+import { Server as HTTPServer } from 'http'
+
+import { Context, ChannelContext, LoguxPatternParams } from './context'
+import { ServerClient } from './server-client'
+
+/**
+ * Logux meta
+ */
+export type LoguxMeta = {
+  id: string
+  time: number
+  subprotocol: LoguxBaseServerOptions['subprotocol']
+  server: string
+  reasons: string[]
+  channels?: string[]
+  clients?: string[]
+  status?: 'waiting' | 'add' | 'clean' | 'processed' | 'subscribed'
+}
+
+/**
+ * Logux base action
+ */
+export type LoguxBaseAction = {
```

Comment:
I think it will be better to have different `LoguxSubscribe`, `LoguxProcessed` and `LoguxUndo` types.

Right now type does not show that `type: 'logux/subscribe' can’t have `id`.